builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -38,6 +39,20 @@ func resourceAwsSecurityGroup() *schema.Resource {
 				},
 			},
 
+			"name_prefix": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) > 100 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 100 characters, name is limited to 255", k))
```

Comment:
I'm just trying a punt in the dark and using a 100 character limit for the name_prefix.